static/js/sandbox-main.js

```diff
@@ -48,3 +48,138 @@ mobileNavTriggerButton.addEventListener('click', function () {
     mobileNavIcon.innerHTML = feather.icons['menu'].toSvg();
   }
 });
+
+// Sanbox Dashboard UI Application List Interactions
+// -------------------------------------------------
+
+// Dashboard App Credential Toggle
+// This shows/hides the client ID and secret for each application
+const appCredentialToggleLinks = document.querySelectorAll('.app-credentials-toggle');
+
+// Toggle on Mouse Click
+appCredentialToggleLinks.forEach((appCredentialToggleLink) => {
+  appCredentialToggleLink.addEventListener('click', function() {
+    let appID = document.querySelector('#id-' + this.id);
+    let appSecret = document.querySelector('#secret-' + this.id);
+    if (appID.type == 'password') {
+      appID.type = 'text';
+      appSecret.type = 'text';
+    } else {
+      appID.type = 'password';
+      appSecret.type = 'password';
+    }
```

Comment:
the implementation of the secret show hide behaviour can be defined separately allowing this function to more explicitly state that it's going to show or hide a secret.

e.g. `appCredentialToggleLink.addEventListener('click', toggleSecret()` or `appCredentialToggleLink.addEventListener('click', function() { toggleSecret(this); })`

Reply:
I think I get what you are saying. And if we extract that out into a `toggleSecret()` function it might be able to also be used in the keyboard navigation event listener, which would clean things up a bit. But I am having trouble preserving variable definitions if I move outside of the context of the `forEach`. I'll work on it some more to see if I can get it working, but any thoughts would be appreciated.

Reply:
Just pushed a commit for this. Moved the contents to a function and then ended up going with the `this.click()` method for the enter keypress.